src/test/kotlin/interactor/GetCityHasMoreSavingsPerMonthInteractorTest.kt

```diff
@@ -19,225 +19,168 @@ internal class GetCityHasMoreSavingsPerMonthInteractorTest {
+        // then
+        assertEquals(cityEntityPrices, getCity)
 
 
     }
     @Test
-    fun should_ReturnNull_When_FoodPriceNull() {
+    fun should_ThrowException_When_FoodPriceNull() {
 
         // given city with food price is null
         every {
-            foodPricesWithNull.foodPrices
-        } returns FoodPrices(10f,20f,null,40f,10f,50f)
-
-        every {
-            foodPricesWithNull.realEstatesPrices
-        } returns RealEstatesPrices(10f,20f,null,40f,10f,50f)
+            cityEntityPrices.foodPrices
+        } returns FoodPrices(null,20f,null,40f,10f,50f)
 
         every {
             dataSource.getAllCitiesData()
-        } returns listOf(foodPricesWithNull)
+        } returns listOf(cityEntityPrices)
+
 
         // when find the most  city in the world have more savings per month.
         citySavings = GetCityHasMoreSavingsPerMonthInteractor(dataSource)
-        val getNull = citySavings.execute()
+        val savingCityExecutable = Executable { citySavings.execute()}
+
 
 
-        // done
-        assertNull(getNull)
+        // then it throws an exception
+        assertThrows(Exception::class.java,savingCityExecutable)
 
 
     }
     @Test
-    fun should_ReturnNull_When_ApartmentNull() {
+    fun should_ThrowException_When_ApartmentNull() {
 
         // given city with apartment is null
         every {
-            apartmentPricesWithNull.foodPrices
+            cityEntityPrices.foodPrices
         }returns  FoodPrices(10f,20f,30f,15f,25f,40f)
 
         every {
-            apartmentPricesWithNull.realEstatesPrices
+            cityEntityPrices.realEstatesPrices
         } returns RealEstatesPrices(10f,20f,null,null,10f,15f)
 
+        every {
+            cityEntityPrices.averageMonthlyNetSalaryAfterTax
+        }returns null
+
         every {
             dataSource.getAllCitiesData()
-        } returns listOf( apartmentPricesWithNull)
+        } returns listOf( cityEntityPrices)
 
 
         // when find the most  city in the world have more savings per month.
         citySavings = GetCityHasMoreSavingsPerMonthInteractor(dataSource)
-        val getNull = citySavings.execute()
+        val savingCityExecutable = Executable { citySavings.execute()}
 
 
-        // done
-        assertNull(getNull)
+        //  then it throws an exception
+        assertThrows(Exception::class.java,savingCityExecutable)
 
 
     }
 
     @Test
-    fun should_ReturnNull_When_AverageSalaryNull() {
+    fun should_ThrowException_When_AverageSalaryNull() {
 
         // given city with average salary  null
         every {
-            averageSalaryWithNull.foodPrices
+            cityEntityPrices.foodPrices
         }returns  FoodPrices(10f,20f,30f,15f,25f,40f)
 
         every {
-            averageSalaryWithNull.realEstatesPrices
+            cityEntityPrices.realEstatesPrices
         } returns RealEstatesPrices(10f,20f,null,null,10f,15f)
```

Comment:
When you're testing the case of NullAverageSalary, Appartments prices shouldn't be null

Reply:
done!